### DISCUSSION TITLE
Pi Pico Board revisions

### DISCUSSION THREAD

**Discussion by Overlights** [terjeio/GRBL_MPG_DRO_BoosterPack#21] [Ideas]
_I state:
Perhaps this is not the right place to share new reviews and for this I apologize to Terje.
GitHub is not an easy place to understand, especially for a non-programmer.
Forking a repository and re-editing it seems like a simple operation.
But understanding to what extent it can be modified (what to keep, what to eliminate, what to change) 
or how it would be "right" to modify it, are different things.
I think it will take some time, but I will appreciate any suggestions._

### Revision Notes
The revisions of Terje's Pi Pico Board that I will describe shortly follows the creation of the Breakout Board for the KeyMatrix, this one:
https://github.com/terjeio/GRBL_MPG_DRO_BoosterPack/discussions/19#discussioncomment-8266420
Designed to be able to interface the assigned buttons more freely, even without using KeyPads, which are usually quite expensive and restrictive from the point of view of the position and choice of keys.

Regarding the Board for Pi Pico, I was curious to start some tests and I would have directly printed the version 1.0 released by Terje, if it had not been for the footprint of the DB15 connector, which partially overlapped one of the fixing holes.
Personally, I find the DB15 connector to be a great choice for remote controls like this.
it is robust, versatile, allows a solid (screw) connection and is relatively inexpensive.
So I started by repositioning the DB15 connector, then I decided to add the double Pad for the LED resistors (SMD and THT), but it was necessary to move the MSP430 and almost all the tracks. 
I took advantage of this to redistribute the spaces between the various connectors.
The result is revision 0.1a:

![GRBL_MPG_DRO_rev_01a_Front](https://github.com/terjeio/GRBL_MPG_DRO_BoosterPack/assets/153271046/bf22d82f-826c-4d9b-8d55-7234c4950a75)

Below, you can see it aligned with the KeyMatrix Breakout Board, positioned so as to have the DB15 connector at the top (the ideal position, in my opinion).

![GRBL_MPG_DRO_rev_01a_GMD-BB](https://github.com/terjeio/GRBL_MPG_DRO_BoosterPack/assets/153271046/aa1fceec-bf30-4caf-a9fc-937d932f19f5)

But I didn't like the upside down texts very much...
So I made a version with the silkscreens rotated 180° and realigned to their respective connectors/components.
Revision 01b:

![GRBL_MPG_DRO_rev_01b_Front_3a](https://github.com/terjeio/GRBL_MPG_DRO_BoosterPack/assets/153271046/525ed73b-3176-437d-8425-449e51a3da17)

Finally, speaking with Phil Barrett, it became clear that it could be interesting to integrate the BB directly into the board.
In this case I used standard 2,54 mm. Pin Headers.
This is the latest revision, 0.1C

![GRBL_MPG_DRO_rev_01C_Front_3v](https://github.com/terjeio/GRBL_MPG_DRO_BoosterPack/assets/153271046/9fd62229-32af-4c33-b6c5-7b0218e0ad20)

The board measures 118 x 68mm.
I also added silkscreen notes on the regulator pins (U3).
The TSR 1-2450 is certainly a good regulator, but it is not exactly cheap (around €7).
The silk-screen printing of the pins: "Vin, Vout, GND", is useful if you want to try one of the many modules: "Step-Down Buck Converter" available on Aliexpress, which allow you to set a fixed output at +5V.

![GRBL_MPG_DRO_rev_01C_Back_3v](https://github.com/terjeio/GRBL_MPG_DRO_BoosterPack/assets/153271046/53a12216-49cc-4d8d-8e7d-549526703e8b)

I exported a test Gerber file from version 0.1C, which is seen correctly by JLCPCB.
But I still have to prepare the final list with component references for JLCPCB, necessary for direct assembly of SMD components.
This is the circuit diagram from the latest revision:
[RP2040_MPG_DRO_Rev_01C_Schematic.pdf](https://github.com/terjeio/GRBL_MPG_DRO_BoosterPack/files/14280731/RP2040_MPG_DRO_Rev_01C_Schematic.pdf)

**Comment by terjeio** [comment]
Nice. The DB15 connector should be replaced with DB9 or some other connector? USB? If running in UART mode with the keypad plugin only four pins are needed. GND, +V, UART RX and TX.

**Comment by Overlights** [comment]
Hi Terje,
Thank you.
And yes, the DB15 connector could easily be replaced by a DB9, but losing at least one GPIO.
By eliminating the I2C connection, you would be left (in no particular order):

1 - GND
2 - TxD
3 - RxD
4 - +12V (or +5V)
5 - CycleStart
6 - FeedHold
7 - Reset
8 - GPIO0
9 - GPIO1

Does this seem correct to you?

If losing control of a GPIO is not a problem, a DB9 connector is certainly more compact than the DB15.
But even if I could only use 4 pins, I wouldn't use a USB connector, which I find more suitable for "Static" connections, where the connected devices are immobile. When you use the pendant, you are focused on the work of the machine. The cable will bend and follow the operator's movements, but the connector must not be able to slide out under any circumstances.
For this use, I think D-SUB connectors are still the best in terms of solidity, easy availability and reasonable cost.

_**If you agree on the 9 pins to keep (as above), I will prepare a new revision with DB9 connector.**_

But I will do my first tests with the rev. 0.1C, due to the fact that I ordered a test batch from JLCPCB.
Curiously, uploading the CPL file (The map of the positions of all the SMD components) in *.csv format led to a preview with the components a few meters away from the board.
I checked everything again (no errors), but repeating the operation always gave the same problem.
The last attempt was to convert the CLP file from *.csv to *.xls.
In this way the import was successful, but I had to change the rotation of some ICs to recover the correct alignment.
I think I will receive them in two or three days.

The first step will be to verify that it works correctly.
_**For this reason, without haste, could you prepare a short procedure describing how to prepare the Firmware for installation?**_

If you agree, I'd like to make some kind of "how to" guide, which describes the steps needed to implement the system and how to customize its features (as I understand and learn how to do it).
In my case, I will have to interface it with the two GRBLHAL Controllers that I have prepared (PicoCNC and T41U5XBB), for two CNC machines that I have in progress, but for the first one to be assembled, it will still take a few months.
In the meantime, it will be interesting to "play" with the design of this Control, perhaps opening a specific discussion to compare the ergonomics of different layouts, from the simplest and most minimal to the most complex.

**Comment by M10CUBE** [comment]
Hi @terjeio , @Overlights ,
After been a long time away I managed to find some time to came back.
Having see the excellent designs @terjeio is given us and his willing to remake it with 2 Picos (if he finds any free time) , the excellent efforts of @Overlights, my old proposal for the MPG
https://gitlab.com/m10cube/m10/-/tree/master/ELECTRICAL/HMI?ref_type=heads
, and most important what others made so far, I think it is time to start thinking a totally new design.
The nice thing is our commitment  in designing with KiCAD.
I if you agree I am wondering for the best place to do that. 
![remote front](https://github.com/terjeio/GRBL_MPG_DRO_BoosterPack/assets/14018762/86311d8b-706a-481d-b431-5b80e253d4d7)
![remote back](https://github.com/terjeio/GRBL_MPG_DRO_BoosterPack/assets/14018762/15dee869-c5a7-4304-b7ed-b8e046a711b6)
![mpg  prototype](https://github.com/terjeio/GRBL_MPG_DRO_BoosterPack/assets/14018762/959e07e4-aeed-449b-b150-5b95bfd16a55)
